protocols/gossipsub/src/behaviour.rs

```diff
@@ -2913,11 +2921,6 @@ where
         connection_id: &ConnectionId,
         endpoint: &ConnectedPoint,
     ) {
-        // Ignore connections from blacklisted peers.
```

Comment:
Can you expand on why we no longer ignore blacklisted peers?

Reply:
I found that in many cases we are under the assumption that any connected peer exists in `connected_peers` and their topic subscriptions exist in `peer_topics`. When peers disconnect we use these mappings to correctly remove them from the mesh and other mappings in the behaviour. If there is any inconsistencies, gossipsub panics (this PR fixes on such instance). 

Although I haven't seen a panic from blacklisted peers (I'm not using them atm), i'm concerned about inconsistencies coming from states where users at arbitrary times blacklist and unblacklist (via the public functions) peers. 

This change represents a safer approach where we register the connections of all peers, add them to the mappings (so they exist regardless of when the user ad-hoc add/removes them) but we still block all messages for whatever peers are blacklisted at the time we receive them. On disconnects, as all peers connections are registered they are removed from all mappings correctly, regardless of the current state of blacklisting.

There probably is a way we could do this without registering the connection, but I think there a quite a few edge cases and this makes it simpler to reason about.

Reply:
Thanks @AgeManning for the details!